### PR TITLE
Revert commit b7ae96f as the bitshares/bitshares-ui#1338 is complete

### PR DESCRIPTION
**Pull request by jmjatlanta**
The issue bitshares/bitshares-ui#1338 is now complete. Now defaulting the parameter ``enable_subscribe_to_all`` to ``false``

A bit of history: This parameter was documented incorrectly, which caused the ui to send ``true``, which in turn caused the API server to spam the clients with unwanted notifications. Now that the bitshares-ui has been corrected, we can default this parameter to ``false`` and avoid sending these unwanted notifications.

PR for #752.

### CODE REVIEW

**Comment by jmjatlanta** [comment]
What steps can we take to minimize the impact? We'll have it in the release notes, but I'd imagine most ui users could care less about core changes.

We could delay this change to a later release. But I don't know if we want to do that or not.

Opinions? Other ideas?

**Comment by abitmore** [comment]
Let's do it in next release. Need to resolve conflicts now.

File: libraries/app/application.cpp [hunk 1]
```diff
@@ -943,12 +942,11 @@ void application::set_program_options(boost::program_options::options_descriptio
          ("api-access", bpo::value<boost::filesystem::path>(), "JSON file specifying API permissions")
          ("plugins", bpo::value<string>(), "Space-separated list of plugins to activate")
          ("io-threads", bpo::value<uint16_t>()->implicit_value(0), "Number of IO threads, default to 0 for auto-configuration")
+         ("enable-subscribe-to-all", bpo::value<bool>()->implicit_value(false),
```
Comment (abitmore) [inline libraries/app/application.cpp]: I think the implicit value should be `true` which is more natural.

File: libraries/app/include/graphene/app/application.hpp [hunk 2]
```diff
@@ -38,7 +38,7 @@ namespace graphene { namespace app {
    class application_options
    {
       public:
-         bool enable_subscribe_to_all = false;
+         bool enable_subscribe_to_all = true;
```
Comment (abitmore) [inline libraries/app/include/graphene/app/application.hpp]: Don't change this one to `true`. 

Sorry, my above comment was not clear enough. Expected behavior:
* this option is disabled by default (if not specified in config.ini nor startup parameters);
* if a user specified it in parameters or config.ini, if without `=X` (only `--enable-subscribe-to-all` but not `--enable-subscribe-to-all=true`), consider the value is `true`.

**Comment by clockworkgr** [comment]
If this was a dynamically typed language this is exactly the sort of behaviour I'd expect if the value of enable-subscribe-to-all was stored as a string and then evaluated as a Boolean.


Seeing as this is C++, I have no idea

**Comment by abitmore** [comment]
I guess @clockworkgr is correct. For example, the `partial-operations` option generated in config.ini is `1` but not `true`. Actually I don't remember if I've ever tried to set it to `false`.

File: libraries/app/application.cpp [hunk 3]
```diff
@@ -410,7 +410,7 @@ void application_impl::startup()
    }
 
    if( _options->count("enable-subscribe-to-all") )
-      _app_options.enable_subscribe_to_all = _options->at("enable-subscribe-to-all").as<bool>();
```
Comment (abitmore) [inline libraries/app/application.cpp]: Sorry, I didn't get it, what's wrong with `_options->at("enable-subscribe-to-all").as<bool>();`?

Comment (jmjatlanta) [reply libraries/app/application.cpp]: If you don't add the command line option, you get an index out of range error

**Comment by jmjatlanta** [comment]
Okay, I have fixed the config.ini problem. It was an error in my logic, as @abitmore pointed out. Now the command line and config.ini file work as expected. A 1/0 or true/false can work in config.ini as expected.

File: libraries/app/application.cpp [hunk 4]
```diff
@@ -409,8 +409,8 @@ void application_impl::startup()
       _force_validate = true;
    }
 
-   if( _options->count("enable-subscribe-to-all") )
-      _app_options.enable_subscribe_to_all = true;
+   if (_options->count("enable-subscribe-to-all"))
```
Comment (abitmore) [inline libraries/app/application.cpp]: Why change the white-spaces? :P

Comment (jmjatlanta) [reply libraries/app/application.cpp]: :-P

**Pull request merged**